Logical properties - support for RTL

ludoboludo opened the pull request:
**PR Summary:** 

This PR changes lots of our CSS so that it uses logical properties instead. 

**Why are these changes introduced?**

This is introduced to make our theme work both in a Left To Right and Right To Left situation. This way merchants using languages that reads from right to left can use our theme. At the moment if a merchant needs to change the reading direction they will most likely contact support to be told that it's not a change we support. 

We used to support that kinda change but on Dawn we're not leveraging any framework or existing library. So when changing the website's reading direction the changes necessary are quite big. 

Logical properties are meant to solve this issue which is why it'd be great to use them going forward. 

**What approach did you take?**

Here is a recap of what they are: 
|  New property | Old one   |
|---|---|
| block-size | height |
| inline-size | width |
| border-block-start | `border-top` |
| border-block-end | `border-bottom` |
| border-inline-start | `border-left` |
| border-inline-end | `border-right` |
| margin-block-start | `margin-top` |
| margin-block-end | `margin-bottom` |
| margin-inline-start | `margin-left` |
| margin-inline-end | `margin-right` |
| padding-block-start | `padding-top` |
| padding-block-end | `padding-bottom` |
| padding-inline-start | `padding-left` |
| padding-inline-end | `padding-right` |
| inset-block-start | `top` |
| inset-block-end | `bottom` |
| inset-inline-start | `left` |
| inset-inline-end | `right` |
| inline-start | value for `float/clear` |
| inline-end | value for `float/clear` |
| start | value for `text-align: left` |
| end | Value for `text-align: right` |


**Here I focus on using only the `margin-inline`, `padding-inline`, `text-align`, `border-inline` logical properties as they are the one that are supported the most. Going all the way back to iOS 12.2 and being supported at 94% by browsers (based on [caniuse.com](caniuse.com))**

I have not changed some original properties in some occasions: 
- both `left` and `right` properties have the same value applied. So the direction shouldn't matter. 
- in some scenarios with have some absolute positioning where we center things using `transform` so I don't need to tweak the values either. 

**Other considerations**

**Testing steps/scenarios**
 
<details><summary>To test, I added a general setting to enable/disable RTL.</summary>

![](https://screenshot.click/12-25-zr5wv-bs7eu.png) 
</details>

- [ ] Test all sections in both LTR and RTL
- [ ] Test different section settings to make sure it doesn't break any previous behaviour
- [ ] Things that are using an absolute positioning and a `transform: translate(50%)` to center them shouldn't be using logical properties. I might have applied it to some. I caught a few that I fixed already.
- [ ] Test with all global settings coming up (shadows, borders, etc)

Here are a couple sites that use RTL and confirm certain patterns (close icon showing at the top left vs the top right in LTR): https://hotcrown.com/ & https://shlomitofir.com/

**Demo links**

- [Store](https://os2-demo.myshopify.com/?preview_theme_id=128347439126)
- [Editor](https://os2-demo.myshopify.com/admin/themes/128347439126/editor)

**Checklist**
- [ ] Added PR summary for [release notes](https://themes.shopify.com/themes/dawn/styles/default#ReleaseNotes)
- [ ] Notified the [help.shopify.com](https://help.shopify.com) team about updates to theme settings (current contact: Kimli)
- [ ] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [ ] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [ ] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [ ] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [ ] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

andrewetchen commented:
Hey Ludo,

As you're probably already aware, the [probot-CLA](https://user-images.githubusercontent.com/28404165/181069776-4926e5bd-7c45-4f8b-8cd0-9998fcc2c641.png) was recently deprecated and replaced with the [shopify-cla](https://user-images.githubusercontent.com/28404165/181069526-e35e4fd4-a151-45d3-9cfe-babfe4d31829.png) action.

To successfully use the new [shopify-cla](https://user-images.githubusercontent.com/28404165/181069526-e35e4fd4-a151-45d3-9cfe-babfe4d31829.png) status check, this PR will be closed and reopened. Thank you and sorry for the minor inconvenience.

Please refer to the following to for more information:

- https://github.com/Shopify/dawn/pull/1859
- https://github.com/Shopify/dawn/pull/1874

ludoboludo commented on assets/base.css:
```diff
@@ -2759,7 +2853,7 @@ details-disclosure > details {
   letter-spacing: 0.1rem;
   line-height: 1;
-  padding: 0.5rem 1.3rem 0.6rem 1.3rem;
+  padding: 0.5rem 1.3rem 0.6rem;
```
Here I kept the original approach as when the value is the same on the right and left the use of logical properties isn't necessary. 
But open to change it to have everything use logical properties. I'll wait for dev feedback on this.

eugenekasimov replied:
Maybe for consistency we should use the new approach. If in the future left and right are not the same we won't need to fix them.

ludoboludo commented:
>Edit: So my action plan going forward is going to remove any mention of inset-inline and replace it with some specific selectors to tweak the values: html[dir="rtl"] .selectors and we can sync later with the other devs to see if that's ok or too much extra code and should be in its own file.
I will also change every shorthand I used.
When that's done, it's potentially in a good spot for reviews and making it into our theme. 👍

This is done. 

[ ] I just rebased with the latest changes we added so I can try and go through the latest CSS added and see if there are new instances where it needs to be applied.

eugenekasimov commented on assets/collapsible-content.css:
```diff
@@ -41,8 +41,8 @@
 .collapsible-content-wrapper-narrow {
   margin: 0 auto;
-  padding-right: 1.5rem;
   padding-left: 1.5rem;
+  padding-right: 1.5rem;
```
I don't think these affect functionality but maybe it's still worth it to change them to `-inline-start` and `-end` for consistency?

eugenekasimov commented on config/settings_schema.json:
```diff
@@ -1508,5 +1508,17 @@
+        "id": "enable_rtl",
+        "label": "t:settings_schema.reading_direction.label",
+        "default": false
+
```
```suggestion
```

eugenekasimov reviewed (commented):
All look and work good. Left a few minor comments.

metamoni commented on assets/component-menu-drawer.css:
```diff
@@ -50,6 +56,16 @@ menu-drawer > details[open] > summary::before {
+}
+
+html[dir="rtl"].js .menu-drawer__submenu {
+  transform: translateX(-100%);
```
These have changed, but the one above (line 39) hasn't. Is this intentional?

metamoni commented on assets/component-menu-drawer.css:
```diff
@@ -169,24 +193,35 @@ details[open].menu-opening > .menu-drawer__submenu {
 }
 
+html[dir="rtl"] .menu-drawer__menu-item > .icon-arrow {
+  transform: translateY(-50%) rotate(180deg);
```
Your comment said we shouldn't be using logical properties on elements using `transform: translate(50%)`. Is this one of them?

metamoni reviewed (commented):
This is very difficult to test without being able to switch to a RTL language and see what things might look like. That being said, I'm confident it's a step in the right direction and we can fix smaller issues as we go along.

I don't know much about what the expected behaviour is with these types of languages, so I think it's really important to get some input from UX or someone who knows more about the topic. At the moment I worry that, if this is not done the right way, we might be causing more confusion rather than helping our users.

But as long as UX or someone with more knowledge confirms that we're not breaking anything major, I'm happy to approve this PR.

ludoboludo commented on assets/component-cart-drawer.css:
```diff
@@ -11,6 +11,11 @@
+  left: initial;
+  right: 0;
+}
+
```
Yeah I'm not 100% sure. Having some folks hep us with this type of thing will be helpful. 
When I look at https://hotcrown.co.il/ they have the cart drawer coming up on the left 🤷

ludoboludo commented on assets/section-main-product.css:
```diff
@@ -27,23 +27,35 @@
 
   .product__info-wrapper {
-    padding: 0 0 0 5rem;
+    padding-top: 0;
+    padding-bottom: 0;
+    padding-inline-start: 5rem;
+    padding-inline-end: 0;
```
Yeah my thought would potentially start changing our language in the editor. But we would need to find the right language that isn't confusing. Otherwise like you said it might be painful to adapt to the theme settings and not be able to use the flexibility of logical properties.
